src/main/kotlin/eu/europa/ec/eudi/pidissuer/security/DPoPTokenReactiveAuthenticationManager.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2023 European Commission
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.europa.ec.eudi.pidissuer.security
+
+import com.nimbusds.oauth2.sdk.dpop.JWKThumbprintConfirmation
+import com.nimbusds.oauth2.sdk.dpop.verifiers.AccessTokenValidationException
+import com.nimbusds.oauth2.sdk.dpop.verifiers.DPoPIssuer
+import com.nimbusds.oauth2.sdk.dpop.verifiers.DPoPProtectedResourceRequestVerifier
+import com.nimbusds.oauth2.sdk.dpop.verifiers.InvalidDPoPProofException
+import com.nimbusds.oauth2.sdk.id.ClientID
+import com.nimbusds.oauth2.sdk.token.DPoPAccessToken
+import com.nimbusds.oauth2.sdk.util.JSONObjectUtils
+import net.minidev.json.JSONObject
+import org.springframework.security.authentication.ReactiveAuthenticationManager
+import org.springframework.security.core.Authentication
+import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException
+import org.springframework.security.oauth2.core.OAuth2TokenIntrospectionClaimNames
+import org.springframework.security.oauth2.server.resource.introspection.BadOpaqueTokenException
+import org.springframework.security.oauth2.server.resource.introspection.SpringReactiveOpaqueTokenIntrospector
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+/**
+ * [ReactiveAuthenticationManager] implementing DPoP authentication.
+ * Implementation information can be found
+ * [here](https://connect2id.com/products/nimbus-oauth-openid-connect-sdk/examples/oauth/dpop#rs).
+ */
+class DPoPTokenReactiveAuthenticationManager(
+    private val introspector: SpringReactiveOpaqueTokenIntrospector,
+    private val verifier: DPoPProtectedResourceRequestVerifier,
+) : ReactiveAuthenticationManager {
+
+    /**
+     * Performs DPoP authentication.
+     */
+    override fun authenticate(authentication: Authentication): Mono<Authentication> =
+        Mono.just(authentication)
```

Comment:
Can we remove the Mono implementation of the method?
Instead perhaps we can use the `mono()` method that accepts a suspended lamda